internal/services/synapse/synapse_pipeline_resource.go

```diff
@@ -0,0 +1,353 @@
+	}
+
+	return nil
+}
+
+func expandSynapseParameters(input map[string]interface{}) map[string]*artifacts.ParameterSpecification {
+	output := make(map[string]*artifacts.ParameterSpecification)
+
+	for k, v := range input {
+		output[k] = &artifacts.ParameterSpecification{
+			Type:         artifacts.ParameterTypeString,
+			DefaultValue: v.(string),
+		}
+	}
+
+	return output
+}
+
+func flattenSynapseParameters(input map[string]*artifacts.ParameterSpecification) map[string]interface{} {
+	output := make(map[string]interface{})
+
+	for k, v := range input {
+		if v != nil {
+			// we only support string parameters at this moment
+			val, ok := v.DefaultValue.(string)
+			if !ok {
+				log.Printf("[DEBUG] Skipping parameter %q since it's not a string", k)
+			}
+
+			output[k] = val
+		}
+	}
+
+	return output
+}
+
+func flattenSynapseAnnotations(input *[]interface{}) []string {
+	annotations := make([]string, 0)
+	if input == nil {
+		return annotations
+	}
+
+	for _, annotation := range *input {
+		val, ok := annotation.(string)
+		if !ok {
+			log.Printf("[DEBUG] Skipping annotation %q since it's not a string", val)
+		}
+		annotations = append(annotations, val)
+	}
+	return annotations
+}
+
+func expandSynapseVariables(input map[string]interface{}) map[string]*artifacts.VariableSpecification {
+	output := make(map[string]*artifacts.VariableSpecification)
+
+	for k, v := range input {
+		output[k] = &artifacts.VariableSpecification{
+			Type:         artifacts.VariableTypeString,
+			DefaultValue: v.(string),
+		}
+	}
+
+	return output
+}
+
+func flattenSynapseVariables(input map[string]*artifacts.VariableSpecification) map[string]interface{} {
+	output := make(map[string]interface{})
+
+	for k, v := range input {
+		if v != nil {
+			// we only support string parameters at this moment
+			val, ok := v.DefaultValue.(string)
+			if !ok {
+				log.Printf("[DEBUG] Skipping variable %q since it's not a string", k)
+			}
+
+			output[k] = val
+		}
+	}
+
+	return output
+}
+
+func deserializeSynapsePipelineActivities(jsonData string) (*[]artifacts.BasicActivity, error) {
+	jsonData = fmt.Sprintf(`{ "activities": %s }`, jsonData)
+	pipeline := &artifacts.Pipeline{}
+	err := pipeline.UnmarshalJSON([]byte(jsonData))
+	if err != nil {
+		return nil, err
+	}
+	return pipeline.Activities, nil
+}
+
+func serializeSynapsePipelineActivities(activities *[]artifacts.BasicActivity) (string, error) {
+	pipeline := &artifacts.Pipeline{Activities: activities}
+	result, err := pipeline.MarshalJSON()
+	if err != nil {
+		return "nil", err
+	}
+
+	var m map[string]*json.RawMessage
+	err = json.Unmarshal(result, &m)
+	if err != nil {
+		return "", err
+	}
+
+	activitiesJson, err := json.Marshal(m["activities"])
+	if err != nil {
+		return "", err
+	}
+
+	return string(activitiesJson), nil
+}
+
+func suppressJsonOrderingDifference(_, old, new string, _ *pluginsdk.ResourceData) bool {
+	return utils.NormalizeJson(old) == utils.NormalizeJson(new)
```

Comment:
i think this is the same as the built in one from the sdk?

Reply:
No, this is not the same with the built in one. This one ignores the order.

Reply:
it does? seems like the plugin SDK one should ignore order:
```
func SuppressJsonDiff(k, old, new string, d *schema.ResourceData) bool {
	oldMap, err := ExpandJsonFromString(old)
	if err != nil {
		return false
	}

	newMap, err := ExpandJsonFromString(new)
	if err != nil {
		return false
	}

	return reflect.DeepEqual(oldMap, newMap)
}

Reply:
I remember I tried before but it didn't work. Let me try again.